ci/scripts/cpp_build.sh

```diff
@@ -308,10 +313,14 @@ fi
 popd
 
 if [ -x "$(command -v ldconfig)" ]; then
-  if [ -x "$(command -v sudo)" ]; then
-    SUDO=sudo
-  else
+  if [ "$(id --user)" -eq 0 ]; then
```

Comment:
We don't need to use `sudo` with the root user.

manylinux image doesn't provide `/usr/bin/sudo` but `/opt/rh/gcc-toolset-12/root/usr/bin/sudo` exists.